play-ws-standalone/src/main/java/play/libs/ws/StandaloneWSRequest.java

```diff
@@ -44,31 +46,52 @@
      * @param body represented as String
      * @return a promise to the response
      */
-    CompletionStage<? extends StandaloneWSResponse> patch(String body);
+    default CompletionStage<? extends StandaloneWSResponse> patch(String body) {
+        return setMethod("PATCH").setBody(body).execute();
+    }
 
     /**
      * Perform a PATCH on the request asynchronously.
      *
      * @param body represented as JSON
      * @return a promise to the response
      */
-    CompletionStage<? extends StandaloneWSResponse> patch(JsonNode body);
+    default CompletionStage<? extends StandaloneWSResponse> patch(JsonNode body) {
+        return setMethod("PATCH").setBody(body).execute();
+    }
 
     /**
      * Perform a PATCH on the request asynchronously.
      *
      * @param body represented as an InputStream
      * @return a promise to the response
+     *
+     * @deprecated Since 1.0.0. Use {@link #patch(Source)} instead.
      */
-    CompletionStage<? extends StandaloneWSResponse> patch(InputStream body);
+    @Deprecated
+    default CompletionStage<? extends StandaloneWSResponse> patch(InputStream body) {
+        return setMethod("PATCH").setBody(body).execute();
+    }
 
     /**
      * Perform a PATCH on the request asynchronously.
      *
      * @param body represented as a File
      * @return a promise to the response
      */
-    CompletionStage<? extends StandaloneWSResponse> patch(File body);
+    default CompletionStage<? extends StandaloneWSResponse> patch(File body) {
+        return setMethod("PATCH").setBody(body).execute();
+    }
+
+    /**
+     * Perform a PATCH on the request asynchronously.
+     *
+     * @param body the body of request
+     * @return a promise to the response
+     */
+    default <U> CompletionStage<? extends StandaloneWSResponse> patch(Source<ByteString, U> body) {
```

Comment:
These `patch`, `post` and `put` methods for `Source` were added so we can keep this consistent with `setBody` versions.